CharlieBackend.Api/Controllers/MentorsController.cs

```diff
@@ -52,6 +51,21 @@ public async Task<ActionResult> PostMentor(long accountId)
             return createdMentorModel.ToActionResult();
         }
 
+        /// <summary>
+        /// Get filter list of lessons for mentor
+        /// </summary>
+        /// <response code="200">Returned filtered list of lessons for mentor </response>
+        [SwaggerResponse(200, type: typeof(IList<LessonDto>))]
+        [Authorize(Roles = "Mentor")]
+        [HttpPost("lessons")]
+        public async Task<ActionResult> GetLessonsForMentor([FromBody]FilterLessonsRequestDto filterModel)
```

Comment:
You can return Task<LessonsDto> as you don't use other responses than success.
Please read this short article
http://www.binaryintellect.net/articles/f36210f2-db45-42cd-9982-f5cd65a1c9a7.aspx